plugins/orchestrator-backend/src/service/router.ts

```diff
@@ -394,6 +394,46 @@ function setupInternalRoutes(
     },
   );
 
+  // v2
+  routerApi.openApiBackend.register(
+    'retriggerInstance',
+    async (c, req: express.Request, res: express.Response) => {
+      const workflowId = c.request.params.workflowId as string;
+      const instanceId = c.request.params.instanceId as string;
+      const endpointName = 'retriggerInstance';
+      const endpoint = `/v2/workflows/${workflowId}/${instanceId}/retrigger`;
+
+      auditLogger.auditLog({
+        eventName: endpointName,
+        stage: 'start',
+        status: 'succeeded',
+        level: 'debug',
+        request: req,
+        message: `Received request to '${endpoint}' endpoint`,
+      });
+
+      const decision = await authorize(
+        req,
+        orchestratorWorkflowExecutePermission,
+        permissions,
+        httpAuth,
+      );
+      if (decision.result === AuthorizeResult.DENY) {
+        manageDenyAuthorization(endpointName, endpoint, req);
+      }
+
+      await routerApi.v2
+        .retriggerInstance(workflowId, instanceId)
+        .then(result => res.status(200).json(result))
+        .catch(error => {
+          auditLogRequestError(error, endpointName, endpoint, req);
+          res
```

Comment:
@ydayagi I worked hard on unifying the error handling
instead of 
res.status(500).json({ message: error.message || INTERNAL_SERVER_ERROR_MESSAGE });
It should just call (next)
That would move the error handling to the standard backstage error handler and create the error in the structure the UI can handle
The UI is implemented to expect a different structure than {message: <text>}

Reply:
fixed